addons/pkg-maven/ftests/src/main/java/org/commonjava/indy/pkg/maven/content/GroupHostedMetadataRemergedOnPomUploadTest.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (C) 2011-2017 Red Hat, Inc. (https://github.com/Commonjava/indy)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.pkg.maven.content;
+
+import org.commonjava.indy.client.core.IndyClientException;
+import org.commonjava.indy.ftest.core.AbstractContentManagementTest;
+import org.commonjava.indy.ftest.core.category.EventDependent;
+import org.commonjava.indy.model.core.Group;
+import org.commonjava.indy.model.core.HostedRepository;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.io.ByteArrayInputStream;
+
+/**
+ * Check that merged metadata in a group full of hosted repositories is updated when a version ( pom file )
+ * is uploaded to the member.
+ * <br/>
+ * GIVEN:
+ * <ul>
+ *     <li>HostedRepositories A, B</li>
+ *     <li>Group G with HostedRepository members A and B</li>
+ *     <li>HostedRepositories A, B all contain metadata path P</li>
+ *     <li>Each metadata file contains different versions of the same project</li>
+ * </ul>
+ * <br/>
+ * WHEN:
+ * <ul>
+ *     <li>upload a new version (pom) to HostedRepository B</li>
+ *     <li>Metadata path P is requested from Group G <b>after events of membership change have settled</b></li>
+ * </ul>
+ * <br/>
+ * THEN:
+ * <ul>
+ *     <li>Group G's metadata path P should reflect values in HostedRepository B's metadata path P</li>
+ * </ul>
+ */
+public class GroupHostedMetadataRemergedOnPomUploadTest
```

Comment:
We also need another version of this with a Group H that contains Group G. The metadata should be requested from this new group (H) in the new test, to verify that metadata cached in the topmost group is cleared when a nested repository's metadata is changed.